phalcon/db/adapter/pdo.zep

```diff
@@ -85,9 +85,12 @@ abstract class Pdo extends Adapter
 	 */
 	public function connect(descriptor = null)
 	{
-		var username, password, dsnParts, dsnAttributes,
+		var eventsManager,
+			username, password, dsnParts, dsnAttributes,
 			persistent, options, key, value;
 
+		let eventsManager = <ManagerInterface> this->_eventsManager;
```

Comment:
Then we can talk about not having the connect done in the constructor and then doing several checks in the many functions that assume that there will always be a `_pdo` object.